v3/common/parser_test.go

```diff
@@ -170,6 +171,91 @@ func TestParser_ParseLongs__Multiple(t *testing.T) {
 	}
 }
 
+func TestParser_ParseFloats__Single(t *testing.T) {
+	p := new(Parser)
+
+	encoded := []byte{0x40, 0x49, 0x0f, 0xdb}
+
+	value, err := p.ParseFloats(encoded, 1, TestDefaultByteOrder)
+	log.PanicIf(err)
+
+	if reflect.DeepEqual(value, []float32{3.14159265}) != true {
+		t.Fatalf("Encoding not correct (1): %v", value)
+	}
+
+	encoded = []byte{0x40, 0x49, 0x0f, 0xdb, 0x40, 0x2d, 0xf8, 0x54}
+
+	value, err = p.ParseFloats(encoded, 1, TestDefaultByteOrder)
+	log.PanicIf(err)
+
+	if reflect.DeepEqual(value, []float32{3.14159265}) != true {
+		t.Fatalf("Encoding not correct (2): %v", value)
+	}
+}
+
+func TestParser_ParseFloats__Multiple(t *testing.T) {
+	p := new(Parser)
+
+	encoded := []byte{0x40, 0x49, 0x0f, 0xdb, 0x40, 0x2d, 0xf8, 0x54}
+
+	value, err := p.ParseFloats(encoded, 2, TestDefaultByteOrder)
+	log.PanicIf(err)
+
+	if reflect.DeepEqual(value, []float32{3.14159265, 2.71828182}) != true {
+		t.Fatalf("Encoding not correct: %v", value)
+	}
+
+	encoded = []byte{0x3f, 0x80, 0x00, 0x01, 0x40, 0x00, 0x00, 0x00}
+
+	value, err = p.ParseFloats(encoded, 2, TestDefaultByteOrder)
+	log.PanicIf(err)
+
+	expectedResult := []float32{1.0, 2.0}
+
+	for i, v := range value {
+		if math.Nextafter32(v, expectedResult[i]) != expectedResult[i] {
```

Comment:
Improper usage. Should be `if v < expectedResult[i] || v >= math.Nextafter32(expectedResult[i], expectedResult[i] + 1) {`.

Reply:
It doesn't work as expected =/

https://play.golang.org/p/mohBUqfg1zM

```go
package main

import (
	"fmt"
	"math"
)

func main() {
	value := []float32{1.0000001, 2}
	expectedResult := []float32{1.0, 2.0}

	for i, v := range value {
		// nextafter = 1.0000001
		nextafter := math.Nextafter32(expectedResult[i], expectedResult[i]+1)
		if v < expectedResult[i] ||
			v >= nextafter {
                        // produces: Encoding not correct (2): 1.0000001
			fmt.Printf("Encoding not correct (2): %v", v)
		}
	}
}
```

Reply:
typo on 1.0 bytes

![image](https://user-images.githubusercontent.com/6754291/101023478-94ccb980-3551-11eb-83fc-fefa84184bf6.png)